File: DelegationDesignPattern/src/main/java/org/example/example01v1/java/Door.java
package org.example.example01v1.java;

public class Door implements DoorActionListener {
    @Override
    public void closeDoor() {
        System.out.println("Door is closing");
    }

    @Override
    public void openDoor() {
        System.out.println("Door is opening");
    }
}

File: DelegationDesignPattern/src/main/java/org/example/example01v1/kotlin/Door.kt
package org.example.example01v1.kotlin

import org.example.example01v1.java.DoorActionListener

open class Door : org.example.example01v1.kotlin.DoorActionListener {
    override fun closeDoor() {
        println("Door is closing")
    }

    override fun openDoor() {
        println("Door is opening")
    }
}